modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -497,12 +486,12 @@ public Assignment putAssignment(User.ID userID, Application.Name applicationName
                     }
 
                     // Add the assignment to the global list of userAssignments of the user
-                    if (assignment.getStatus() != Assignment.Status.EXPERIMENT_EXPIRED) {
+                    if (Experiment.State.PAUSED != experiment.getState()) { // the Experiment is set to PAUSED in the case of a reached endDate
```

Comment:
@AndreaSuckro This change is also NOT backward compatible, Please check with @calvinwma ... @calvinwma A suggestion, to keep it backward compatible; now we may completely remove this IF condition as we are getting rid of EXPERIMENT_EXPIRED state and we used to add entry in the 'userAssignments' map even in case of Experiment.State.PAUSED